### DISCUSSION TITLE
Kubernetes 1.27 Major Themes

### DISCUSSION THREAD

**Discussion by harshanarayana** [kubernetes/sig-release#2166] [General]
This is a discussion to discuss Major Themes for the v1.27 release.

Here is the [discussion thread](https://github.com/kubernetes/sig-release/discussions/2047) from the last release cycle.

cc @kubernetes/release-team-leads 
cc @AnaMMedina21 @rjsadow @sanchita-07 @yrs147

**Comment by pohly** [comment]
SIG Instrumentation: PRs are under review which convert kube-controller-manager and kube-scheduler to structured, contextual logging. I'll update this comment once I know whether that work gets completed.

The description of https://github.com/kubernetes/enhancements/issues/3077 has a table where I am tracking pending PRs.

**Reply by hwinkel** [reply]
May even open tracing if you touch logging ?

**Comment by ameukam** [comment]
SIG K8S Infra / SIG Release: Freeze promotion of the container images to k8s.gcr.io.

Announcement: https://kubernetes.io/blog/2023/02/06/k8s-gcr-io-freeze-announcement/
Enhancement tracking issue: https://github.com/kubernetes/enhancements/issues/3720

**Comment by saschagrunert** [comment]
#### SeccompDefault graduates to stable

Announcement: _TBD_
Enhancement: https://github.com/kubernetes/enhancements/issues/2413

**Comment by saschagrunert** [comment]
#### Kubelet OpenTelemetry tracing graduates to beta

Announcement: _TBD_
Enhancement: https://github.com/kubernetes/enhancements/issues/2831

cc @sallyom

**Comment by saschagrunert** [comment]
#### DownwardAPIHugePages graduates to stable

Announcement: _TBD_
Enhancement: https://github.com/kubernetes/enhancements/issues/2053

**Comment by xing-yang** [comment]
Here are major themes from SIG Storage

New Features:
- The [Robust VolumeManager Reconstruction](https://github.com/kubernetes/enhancements/issues/3756) feature moved to Beta in v1.27. This is a VolumeManager refactoring that allows kubelet to populate additional information about how existing volumes are mounted during the kubelet startup. In general, this makes volume cleanup more robust.
- The [SELinux Relabeling with Mount Options](https://github.com/kubernetes/enhancements/issues/1710) feature moved to Beta in v1.27. This feature speeds up container startup by mounting volumes with the correct SELInux label instead of changing each file on the volumes recursively.
- The [ReadWriteOncePod PV Access Mode](https://github.com/kubernetes/enhancements/issues/2485) feature moved to Beta in v1.27. This feature introduces a new ReadWriteOncePod access mode for PersistentVolumes that restricts access to a single pod on a single node.

In-tree Driver Removals (planned)
- [CSI Migration for Azure Disk](https://github.com/kubernetes/enhancements/issues/1490) moved to GA in v1.24.  The in-tree Azure Disk plugin is targeted for removal in v1.27.
- [CSI Migration for AWS EBS](https://github.com/kubernetes/enhancements/issues/1487) moved to GA in v1.25. The in-tree AWS EBS plugin is targeted for removal in v1.27.

cc @saad-ali @msau42 @jsafrane

**Comment by aravindhp** [comment]
**SIG Windows** 

New feature:
- [NodeLogQuery](https://github.com/kubernetes/enhancements/issues/2258) has been introduce as an alpha feature. This feature helps cluster administrators debug issues with services running on nodes by allowing them to query service logs. To use the feature, ensure that the `NodeLogQuery` feature gate is enabled for that node, and that the kubelet configuration options `enableSystemLogHandler` and `enableSystemLogQuery` are both set to true.

**Comment by kerthcet** [comment]
From __SIG Scheduling__

New Features:
- [Mutable Pod scheduling directives when gated](https://github.com/kubernetes/enhancements/issues/3838): This is a new feature but starts at beta level. It's similar to `mutable scheduling directives for suspended Jobs`, but for Pod only. With this feature enabled, pod's node affinity and selector is mutable when blocked by a scheduling gate, which gives rights to external controllers to influence the scheduling directives. This is a derivative of `Pod Scheduling Readiness`.

Beta -> GA
- [Mutable scheduling directives for suspended Jobs](https://github.com/kubernetes/enhancements/issues/2926): This was introduced in v1.22 and started as a beta level, now it's stable. It allows to mutate the node scheduling directives, namely the pod template's node affinity, node selector, tolerations, schedulingGates, annotations and labels of the job's pod template, which enables a higher-level queue controller to inject such scheduling directives before un-suspending a job.

Alpha -> Beta:
- [Pod Scheduling Readiness](https://github.com/kubernetes/enhancements/issues/3521): This introduces a `.spec.schedulingGates` field to Pod's API, to indicate whether the Pod is allowed to be scheduled or not. External controllers can use the gates to hold a Pod from scheduling based on their business needs. The `SchedulingGates` plugin is enabled by default now in scheduler framework.
- [Respect PodTopologySpread after rolling upgrades](https://github.com/kubernetes/enhancements/issues/3243): This adds a new field `matchLabelKeys` to podTopologySpread, it only cares about the keys but values which helps to simplify the configuration in pod topology spreading, and provides a solution for unbalance scheduling in pods rolling update.


cc @ahg-g @alculquicondor @Huang-Wei
